src/Redux/missions/missionSlice.js

```diff
@@ -0,0 +1,65 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const initialState = {
+  missions: [],
+  isLoading: false,
+  error: null,
+};
+
+export const getMissions = createAsyncThunk('missions/getMissions', async () => {
+  try {
+    const response = await fetch('https://api.spacexdata.com/v3/missions');
+    const allMission = await response.json();
+    return allMission;
+  } catch (error) {
+    throw new Error('failed to get mission data');
+  }
+});
+
+const missionSlice = createSlice({
+  name: 'mission',
+  initialState,
+  reducers: {
+    onboardMission: (state, action) => {
+      const missionId = action.payload;
+      const newState = { ...state };
+      newState.missions = state.missions.map((mission) => {
+        if (mission.mission_id !== missionId) {
+          return mission;
+        }
+        return { ...mission, reserved: true };
+      });
+      return newState;
+    },
+
+    offboardMission: (state, action) => {
+      const missionId = action.payload;
+      const newState = { ...state };
+      newState.missions = state.missions.map((mission) => {
+        if (mission.mission_id !== missionId) {
+          return mission;
+        }
+        return { ...mission, reserved: false };
+      });
+      return newState;
+    },
+  },
```

Comment:
When you navigate to a different page (rockets or my profile page) and returns to the mission page, all joined missions return to the previous state (unjoined) after joining a mission. Kindly make sure the state is maintained after joining or unjoining a mission unless you change it by clicking the button or reloading the data. :pray: